src/main/java/core/basesyntax/service/operation/SupplyOperation.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.service.operation;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.FruitShopService;
+import core.basesyntax.strategy.OperationHandler;
+
+public class SupplyOperation implements OperationHandler {
+    private final FruitShopService fruitShopService;
+
+    public SupplyOperation(FruitShopService fruitShopService) {
+        this.fruitShopService = fruitShopService;
+    }
+
+    @Override
+    public void processWithTransaction(FruitTransaction transaction) {
+        fruitShopService.supplyFruit(transaction);
+    }
```

Comment:
Please move logic from fruitShopService to handlers like below, there is no need for separate service with such usage
```suggestion
    @Override
    public void processWithTransaction(FruitTransaction transaction) {
        if (Storage.getFruits().containsKey(transaction.getFruit())) {
            int quantity = Storage.getFruits().get(transaction.getFruit());
            Storage.getFruits().put(transaction.getFruit(),transaction.getQuantity() + quantity);
        }
        Storage.getFruits().put(transaction.getFruit(), transaction.getQuantity());
    }
```